src/interactive/azext_interactive/azclishell/recommendation.py

```diff
@@ -0,0 +1,389 @@
+                    return
+            send_feedback(0, trigger_commands, processed_exception, recommendations, rec=None)
+
+    def feedback_scenario(self, scenario_idx, scenario=None):
+        """Send user's command choice in recommendations to telemetry.
+
+        :param scenario_idx: idx of the scenario in scenario recommendations list
+        :param scenario: selected scenario item
+        """
+        if self.cur_thread and not self.cur_thread.is_alive():
+            recommendations = self.cur_thread.result
+            trigger_commands = [cmd['command'] for cmd in self.cur_thread.command_history[-2:]]
+            processed_exception = self.cur_thread.processed_exception
+            if not recommendations or not scenario:
+                send_feedback(-1, trigger_commands, processed_exception, recommendations, rec=None)
+            else:
+                send_feedback(f'b{scenario_idx+1}', trigger_commands, processed_exception, recommendations, rec=scenario)
+
+    def update_executing(self, cmd, feedback=True):
+        """Update executing command info, and prefetch the recommendation result as if the execution is successful
+
+        :param cmd: The command that would be executed soon
+        :param feedback: whether send feedback to telemetry using this command as actual execution vs
+        previous recommendation
+        """
+        if not self.enabled:
+            return
+        # e.g. `az  webapp create --name xxx -g xxxx` => `webapp create`
+        command = re.sub('^az *', '', cmd).split('-', 1)[0].strip()
+        param = sorted([p for p in cmd.split() if p.startswith('-')])
+        if feedback:
+            self._feedback_command(command)
+        self.executing_command = {'command': command, 'arguments': param}
+        self._update()
+
+    def update_exec_result(self, exit_code, result_summary=''):
+        """Update execution result of executing_command set previously, fetch recommendation result if execution fails
+
+        :param exit_code: exit code of previous executing command
+        :param result_summary: Result Summary of previous executing command
+        """
+        if not self.enabled or not self.executing_command:
+            return
+        self.rec_path.append_result(self.executing_command['command'], self.executing_command['arguments'],
+                                    exit_code, result_summary)
+        self.executing_command = None
+        # If the execution of the last command fails, get the recommendation result again
+        if exit_code != 0:
+            self._update()
+
+    def _update(self):
+        """Update recommendation result in new thread"""
+        self.cur_thread = RecommendThread(self.cli_ctx, self.rec_path, self.executing_command,
+                                          self.on_prepared_callback)
+        self.cur_thread.start()
+
+    def _get_result(self, rec_type=RecommendType.Command):
+        if not self.cur_thread:
+            return None
```

Comment:
Please add a comment to explain what `None` means in this case

Reply:
Done